File: algorithms/454.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int fourSumCount(vector<int> &A, vector<int> &B, vector<int> &C, vector<int> &D) {
        int n = A.size();
        unordered_map<int, int> m;
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                m[A[i]+B[j]] += 1;
        int result = 0;
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                result += m[-C[i]-D[j]];
        return result;
    }
};

int main() {
    return 0;
}

File: algorithms/python/454.py
from collections import defaultdict
class Solution(object):
    def fourSumCount(self, A, B, C, D):
        """
        :type A: List[int]
        :type B: List[int]
        :type C: List[int]
        :type D: List[int]
        :rtype: int
        """
        n = len(A)
        m = defaultdict(int)
        for i in range(n):
            for j in range(n):
                m[A[i]+B[j]] += 1
        rls = 0
        for i in range(n):
            for j in range(n):
                rls += m[-C[i]-D[j]]
        return rls
